### PR TITLE
#8 add task component

### PR DESCRIPTION
**Pull request by BojanJakic**
There is a few issues about this ticket :
1. We have do fix CORS problem. I did it for testing purpose so I can push it as part of existing #5 backend ticket or we can make a new one.
2. At the moment I get  java.sql.SQLException: Field 'id' doesn't have a default value, it will be solved after Gosa update script by adding auto-increment to id field

I create pull request so you can take a look what have I done so far.

One question that is not related  only to this ticket, but whole project
Would we keep all of components in components folder, or make additional one(containers) for stateful components ?

### CODE REVIEW

File: src/actions/task_actions.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import {SAVE_TASK} from './type';
+import axios from 'axios';
+
+export const saveTask = newTask => {
+
+    var url = 'http://localhost:8080/tasks';
```
Comment (cctraining) [inline src/actions/task_actions.js]: Use let/const

File: src/components/App.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
 import React, { Component } from 'react';
-
+import Task from './Task'
```
Comment (cctraining) [inline src/components/App.js]: Rename Task.js -> task.js. Then import will be `import Task from './task'`

File: src/components/Task.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import { saveTask } from '../actions/task_actions.js';
+
+class Task extends Component {
+    render = () => {
```
Comment (cctraining) [inline src/components/Task.js]: Can you try to do it without checking the `onKeyUp` and `keyCode` ;)

File: src/components/Task.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    handleKeyPress = (evt) => {
+        if (evt.keyCode === 13) {
+            this.props.dispatchSaveAction(evt.target.value)
+        }
+    }
+}
+
+const mapDispatchToProps = (dispatch) => {
+    return {
+        dispatchSaveAction: newTask => dispatch(saveTask(newTask))
```
Comment (cctraining) [inline src/components/Task.js]: I would call this `saveTask` instead of `dispatchsaveAction`

File: src/reducers/taskReducer.js [hunk 5]
```diff
@@ -1,5 +1,18 @@
 const taskReducer = (state = {}, action) => {
-    return state;
+
+    switch (action.type) {
+        case 'SAVE_TASK' :
```
Comment (cctraining) [inline src/reducers/taskReducer.js]: Import type from `actions/type` ?

**Comment by cctraining** [comment]
@BojanJakic 
- CORS will be fixed. That issue is expected.
- ID in DB should be solved 
- We should use containers/components

File: src/actions/task_actions.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    };
+
+    return (dispatch) => {
+        return axios.post(url, task).then(response => {
+            dispatch({
+                type: SAVE_TASK,
+                payload: response.data
+            });
+
+            form.reset();
```
Comment (nikola-lisac) [inline src/actions/task_actions.js]: I think it's better to put the code for the form (evt.preventDefault, form.reset etc) inside a handler method in the Task component, it keeps our actions cleaner.

**Comment by nikola-lisac** [comment]
I think it's better to put the code for the form (evt.preventDefault, form.reset etc) inside a handler method in the Task component, it keeps our actions cleaner.

File: src/actions/task_actions.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import {SAVE_TASK} from './type';
+import axios from 'axios';
+
+export const saveTask = inputsValue => {
+    let url = 'http://localhost:8080/tasks';
```
Comment (cctraining) [inline src/actions/task_actions.js]: This can be `const url`?

**Pull request merged**